packages/core/src/external/carequality/ihe-gateway-v2/saml-client.ts

```diff
@@ -0,0 +1,115 @@
+import https from "https";
+import axios from "axios";
+import { errorToString } from "../../../util/error/shared";
+import { BulkSignedXCPD } from "../../saml/xcpd/iti55-envelope";
+import { verifySaml } from "../../saml/security/verify";
+import { capture } from "../../../util/notifications";
+
+export async function sendSignedXml({
+  signedXml,
+  url,
+  certChain,
+  publicCert,
+  key,
+  passphrase,
+  trustedRootCert,
+}: {
+  signedXml: string;
+  url: string;
+  certChain: string;
+  publicCert: string;
+  key: string;
+  passphrase: string;
+  trustedRootCert?: string[];
+}): Promise<string> {
+  const agent = new https.Agent({
+    rejectUnauthorized: false,
+    cert: certChain,
+    key: key,
+    passphrase,
+    ca: trustedRootCert,
+  });
+
+  const verified = verifySaml({ xmlString: signedXml, publicCert });
+  if (!verified) {
+    throw new Error("Signature verification failed.");
+  } else {
+    console.log("Signature verification passed.");
```

Comment:
Right now, while in this initial rollout, it will be helpful for debugging to know if the reason things failed is due to some bug in signature verification as opposed to some other piece of this process

Reply:
WIll remove